modules/apps_api/config/routes.rb

```diff
@@ -6,8 +6,9 @@
   match '/v0/healthcheck', to: 'metadata#healthcheck', via: [:get]
 
   namespace :v0, defaults: { format: 'json' } do
-    get 'directory/scopes/:category', to: 'directory#scopes'
-    resources :directory, only: %i[index]
+    get 'directory', to: 'directory#index'
+    get 'directory/scopes/:category', to: 'directory#scopes', as: 'scopes', defaults: { category: 'unknown_category' }
+    get 'directory/:name', to: 'directory#show'
```

Comment:
why not just define the scopes situation here? pretty sure something like this should work, and mean you wouldn't need the if check and redirect in the controller.
```
get 'directory/scopes/:category', to: 'directory#scopes', as: 'scopes', defaults: { category: 'unknown_category' }
get 'directory/scopes', to: 'directory#scopes'
get 'directory/:name', to: 'directory#show'
```
Sidenote, im a bit surprised `default` for category wouldn't work that way by default, but im pretty sure adding that line in the middle to explicitly look for "scopes" above ":name" would remove the need for your controller if/redirect bit

Reply:
It works correctly I just end up having to set the default explicitly again. Not sure which is more ugly tbh. The routing is only ugly cause I have the scopes not being handled in their own controller..
`  get 'directory/scopes/:category', to: 'directory#scopes', defaults: { category: 'unknown_category' }
    get 'directory/scopes', to: 'directory#scopes', defaults: { category: 'unknown_category' }
    resources 'directory', only: %i[index show]`

Reply:
i suppose you could set that default category to a variable above and reference it so they stay in sync.
but whatever you think is best. i typically see basic if/then/redirect in a controller and think it feels wrong, seems like something that should be handled solely in the routes file, but i approved the pr if this way makes more sense to you :+1: